### PR TITLE
[fix](planner) Fix delete statement on MOR unique table

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes
add conditions for mor table removed in https://github.com/apache/doris/pull/26776

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by bobhan1** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DeleteStmt.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ private void constructInsertStmt() throws AnalysisException {
                 expr = new BoolLiteral(true);
             } else if (column.isKey()) {
                 expr = new SlotRef(targetTableRef.getAliasAsName(), column.getName());
-            } else if (!isMow && !column.isVisible()) {
+            } else if (!isMow && (!column.isVisible() || (!column.isAllowNull() && !column.hasDefaultValue()))) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DeleteStmt.java:160]: please add some comment to explain these conditions

Comment (bobhan1) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/DeleteStmt.java:160]: This condition is accidentally deleted in https://github.com/apache/doris/pull/26776 due to not considering mor table.

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**